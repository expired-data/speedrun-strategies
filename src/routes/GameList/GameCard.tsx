import React, { FC } from "react"; 
import styled from "styled-components";

import { BulkGame } from "api/src"; 

export interface Props { 
    game: BulkGame; 
}

const Card = styled.div`
    border: 1px solid #CDCDCD; 
    margin-top: 20px;
`

export const GameCard: FC<Props> = ({ game }) => ( 
    <Card>
        <span>Name: {game.names.international}</span>
        <br />
        <span>Id: {game.id}</span>
    </Card> 
) 