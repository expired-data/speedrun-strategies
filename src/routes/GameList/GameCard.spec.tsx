import React from "react";
import { render } from "@testing-library/react";
import type { BulkGame } from "api/src";
import { GameCard } from "./GameCard";

const MOCK_GAME: BulkGame = {
  abbreviation: "",
  id: "abcdefg",
  names: {
    international: "test game",
    japanese: null,
  },
  weblink: "",
};

test("GameCard renders as expected", () => {
  const { container } = render(<GameCard game={MOCK_GAME} />);
  expect(container.innerHTML).toMatchSnapshot();
});
