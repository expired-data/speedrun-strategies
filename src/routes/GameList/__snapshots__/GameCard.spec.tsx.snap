// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameCard renders as expected 1`] = `"<div class=\\"sc-bdfBwQ grXAio\\"><span>Name: test game</span><br><span>Id: abcdefg</span></div>"`;
